### PR TITLE
Dg ug updates and ppp

### LINKED ISSUE #172: [PE-D] Are commands supposed to be case-sensitive?
It is stated in the UG that only parameters entered by the user are not case-sensitive.

While the format specified in the UG was `/listfav` (`/listfav` is not a parameter entered by a user though...), 

typing `/Listfav` did not throw an error. (for the other commands as well)

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: chuckiex3/ped#4

### PR DESCRIPTION
**Pull request by EthanWong2212**
#55 
#121 
#192 
Fix #172

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -189,23 +189,37 @@ The `AddFavCommand#executeCommand()` method of AddFavCommand Class executes the
...
-<br><br>
+5. `ExecFavCommand#getFav()` is self invoked to obtain the required `Fav` object from `FavList`.
+    - An exception is thrown if there is no `Fav` object in the specified index in `FavList`.
+6. A new `Parser` object is instantiated to run the command in the `Fav` object.
+    - If the command from the Fav object throws an exception when running, the exception will be caught in `ExecFavCommand#executeCommand()` and the `Fav` object will be deleted from the `FavList`.
 
 The following sequence diagram illustrates the steps taken by the program when the user calls the `/execfav` command.
 ![ExecFav_Sequence_Diagram](DG_Diagrams/ExecFavCommand/ExecFavCommand.png)
 
 ####Design Considerations
```
Comment (yuxinwu99) [inline docs/DeveloperGuide.md]: what is this line for?

File: docs/team/EthanWong2212.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+reflects my code contribution.
+
+#### Enhancements implemented   
+
+#### Contributions to documentation
+* Added documentation for features `/route`and `/descfav`.
+
+#### Contributions to the DG
+
+
```
Comment (yuxinwu99) [inline docs/team/EthanWong2212.md:30]: You might want to either remove the empty sections or add something to it.

File: docs/team/EthanWong2212.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#### Contributions to team-based tasks 
+* Skeleton code for favourite features.
+* Maintained the issue tracker for tasks and documentation.
+
+#### Review/mentoring contributions
+
```
Comment (yuxinwu99) [inline docs/team/EthanWong2212.md:36]: This section is empty

File: docs/team/EthanWong2212.md (hunk 2 above)
Comment (EthanWong2212) [reply docs/team/EthanWong2212.md:30]: things will be added in future

File: docs/team/EthanWong2212.md (hunk 3 above)
Comment (EthanWong2212) [reply docs/team/EthanWong2212.md:36]: things will be added in future

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -189,23 +189,37 @@ The `AddFavCommand#executeCommand()` method of AddFavCommand Class executes the
...
+##### Aspect: Choice of command object in FavList to execute
```
Comment (yuxinwu99) [inline docs/DeveloperGuide.md:209]: You can talk about how typing full description is not very user friendly compared to just typing out numbers. Especially how some descriptions can be longer than the command itself.

Comment (EthanWong2212) [reply docs/DeveloperGuide.md:209]: good idea!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (EthanWong2212) [reply docs/DeveloperGuide.md]: styling error solved

**Review by yuxinwu99 — APPROVED** [review]

**Pull request merged**